### DISCUSSION TITLE
What is the design principle of flow ID?

### DISCUSSION THREAD

**Discussion by wenqi73** [ory/kratos#2263] [Q&A] [answered]
Curious why the flow ID is needed. Could someone tell me what flow ID does?

**Comment by vinckr** [comment, marked as answer]
The flow ID is used to identify each flow and make sure man-in-the-middle attacks are not possible. There are probably other considerations going into it.

For reference when I login at Ory Network
e.g. `https://console.ory.sh/login?flow=9e42c3ce-da4b-42b2-846d-19eb9143d086`

The `9e42c3ce-da4b-42b2-846d-19eb9143d086` is the ID for this particular login flow.
It looks the same for any other Ory Kratos deployment.

**Reply by vinckr** [reply]
Hey @wenqi73 ,
Just checking in if your question was answered or if there was still something unclear.
I will mark this as answered for now, but feel free to come back to the discussion any time.

Thanks a ton! 🐝

**Reply by dmitry-kostin** [reply]
Could you please provide further details on this matter, or point us towards relevant resources? Theoretically, a man-in-the-middle can observe the flow ID, so from this viewpoint, nothing appears to change. This may seem like a dumb question, but I am not an expert in security, and I believe this information could be useful to others as well. Thanks in advance.
